internal/acctest/oauthclient/oauth_client_test.go

```diff
@@ -0,0 +1,460 @@
+	optionalHcl := ""
+	optionalRedirectUris := ""
+	if resourceModel.redirectUris != nil {
+		optionalRedirectUris = fmt.Sprintf(`
+		redirect_uris = %s
+		`, acctest.StringSliceToTerraformString(resourceModel.redirectUris))
+	}
+	if resourceModel.includeOptionalAttributes {
+		optionalHcl = fmt.Sprintf(`
+		enabled = %t
+		bypass_approval_page = %t
+		description = "%s"
+		logo_url = "%s"
+		allow_authentication_api_init = %t
+		require_pushed_authorization_requests = %t
+		require_jwt_secured_authorization_response_mode = %t
+		restrict_scopes = %t
+		restricted_scopes = %s
+		restricted_response_types = %s
+		restrict_to_default_access_token_manager = %t
+		validate_using_all_eligible_atms = %t
+		%s
+		%s
+		%s
+		require_proof_key_for_code_exchange = %t
+		ciba_delivery_mode = "%s"
+		ciba_polling_interval = %d
+		ciba_require_signed_requests = %t
+		ciba_user_code_supported = %t
+		ciba_notification_endpoint = "%s"
+		jwt_secured_authorization_response_mode_encryption_algorithm = "%s"
+		jwt_secured_authorization_response_mode_content_encryption_algorithm = "%s"
+		token_introspection_signing_algorithm = "%s"
+		require_signed_requests = %t
+		`,
+			*resourceModel.enabled,
+			*resourceModel.bypassApprovalPage,
+			*resourceModel.description,
+			*resourceModel.logoUrl,
+			*resourceModel.allowAuthenticationApiInit,
+			*resourceModel.requirePushedAuthorizationRequests,
+			*resourceModel.requireJwtSecuredAuthorizationResponseMode,
+			*resourceModel.restrictScopes,
+			acctest.StringSliceToTerraformString(resourceModel.restrictedScopes),
+			acctest.StringSliceToTerraformString(resourceModel.restrictedResponseTypes),
+			*resourceModel.restrictToDefaultAccessTokenManager,
+			*resourceModel.validateUsingAllEligibleAtms,
+			oidcPolicyHcl(resourceModel.oidcPolicy),
+			clientAuthHcl(resourceModel.clientAuth),
+			jwksSettingsHcl(resourceModel.jwksSettings),
+			*resourceModel.requireProofKeyForCodeExchange,
+			*resourceModel.cibaDeliveryMode,
+			*resourceModel.cibaPollingInterval,
+			*resourceModel.cibaRequireSignedRequests,
+			*resourceModel.cibaUserCodeSupported,
+			*resourceModel.cibaNotificationEndpoint,
+			*resourceModel.jwtSecuredAuthorizationResponseModeEncryptionAlgorithm,
+			*resourceModel.jwtSecuredAuthorizationResponseModeContentEncryptionAlgorithm,
+			*resourceModel.tokenIntrospectionSigningAlgorithm,
+			*resourceModel.requireSignedRequests,
+		)
+	}
+
+	return fmt.Sprintf(`
+resource "pingfederate_oauth_client" "%s" {
+  client_id   = "%s"
+  grant_types = %s
+  name        = "%s"
+	%s
+	%s
+}`, resourceName,
+		oauthClientId,
+		acctest.StringSliceToTerraformString(resourceModel.grantTypes),
+		resourceModel.name,
+		optionalRedirectUris,
+		optionalHcl,
+	)
+}
+
+// Test that the expected attributes are set on the PingFederate server
+func testAccCheckExpectedOauthClientAttributes(config oauthClientResourceModel, testOptionalAttributes bool) resource.TestCheckFunc {
```

Comment:
You could read the "includeOptionalAttributes" property from the model rather than using a separate boolean parameter in this function.